src/Streamly/Internal/Network/Socket.hs

```diff
@@ -106,25 +105,27 @@ import qualified Streamly.Internal.Data.Array.Foreign.Type as A
 import qualified Streamly.Internal.Data.Stream.IsStream as S
 import qualified Streamly.Internal.Data.Stream.StreamD.Type as D
 
--- | @'handleWithM' socket act@ runs the monadic computation @act@ passing the
--- socket handle to it.  The handle will be closed on exit from 'handleWithM',
--- whether by normal termination or by raising an exception.  If closing the
--- handle raises an exception, then this exception will be raised by
--- 'handleWithM' rather than any exception raised by 'act'.
+-- | @'forSocketM' action socket@ runs the monadic computation @action@ passing
+-- the socket handle to it.  The handle will be closed on exit from
+-- 'forSocketM', whether by normal termination or by raising an exception.  If
+-- closing the handle raises an exception, then this exception will be raised
+-- by 'forSocketM' rather than any exception raised by 'action'.
 --
--- @since 0.7.0
-{-# INLINE handleWithM #-}
-handleWithM :: (MonadMask m, MonadIO m) => (Socket -> m ()) -> Socket -> m ()
-handleWithM f sk = finally (f sk) (liftIO (Net.close sk))
+-- @since 0.8.0
+{-# INLINE forSocketM #-}
+forSocketM :: (MonadMask m, MonadIO m) => (Socket -> m ()) -> Socket -> m ()
+forSocketM f sk = finally (f sk) (liftIO (Net.close sk))
```

Comment:
`forSocketM` sounds like a loop.

Reply:
`for` is the opposite of `with`, the confusion may be because we are used to `for` as a loop. Other alternative is `useSocketM`, any other better/short alternatives?